_core/smart/tutorial.md

```diff
@@ -0,0 +1,335 @@
+   (let ((balance (get-balance tx-sender)))
+     (token-credit! tx-sender amount)))
+
+(token-credit! 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR 10000)
+(token-credit! 'SM2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQVX8X0G 300)
+```
+
+The final two lines of the program pass a principal, represented by a Stacks address, and an amount to the private user-defined `token-credit` function.
+
+Smart contracts may call other smart contracts using a `contract-call!` function. This means that if a transaction invokes a function in a given smart contract, that function is able to make calls into other smart contracts on your behalf. The ability to read and do a static analysis of Clarity code allows clients to learn which functions a given smart contract will ever call. Good clients should always warn users about any potential side effects of a given transaction.
+
+Take a moment to `cat` the contents of the `name.clar` file.
+
+```bash
+cat names.clar
+````
+
+Which `tokens.clar` function is being called? 
+
+## Task 3: Initialize data-space and launch contracts
+
+In this task, you interact with the the contracts using the `clarity-cli` command line. 
+
+1. Initialize a new `db` database in the `/data/` directory
+
+    ```bash
+    #  clarity-cli initialize /data/db
+    Database created
+    ```
+
+    You should see a message saying `Database created`. The command creates an SQLlite database.  The database is available in the container and also in your workstation. In this tutorial, your workstation mount should at this point contain the `$HOME/blockstack-dev-data/db`  directory.
+
+2. Type check the `names.clar` contract.
+
+    ```bash
+    #  clarity-cli check sample-programs/names.clar /data/db
+    ```
+    You should get an error:
+
+    ```
+    Type check error.
+    NoSuchContract("tokens")
+    ```
+
+    This happens because the `names.clar` contract _calls_ the `tokens.clar` contract, and that contract has not been created on the blockchain.
+
+3. Type check the `tokens.clar` contract, it should pass a check as it does not use the `contract-call` function:
+
+    ```bash
+    # clarity-cli check sample-programs/tokens.clar /data/db
+    ```
+
+    When the `check` command executes successfully, it does not output any information, it just exits with the stand UNIX `0` exit code.
+
+4. Launch the `tokens.clar` contract.
+
+   You use the `launch` command to instantiate a contract on the Stacks blockchain. If you have dependencies between contracts, for example names.clar is dependent on tokens.clar, you must launch the dependency first.
+
+    ```bash
+    # clarity-cli launch tokens sample-programs/tokens.clar /data/db
+    Contract initialized!
+    ```
+    
+    Once launched, you can execute the contract or a public method on the contract. Your development database has an instantiated `tokens` contract. If you were to close the VM, you and relaunch it later with the same mount point and you wouldn't need to relaunch.
```

Comment:
> If you were to close the VM, you and relaunch it later with the same mount point and you wouldn't need to relaunch.

Could be:

If you were to close the VM and relaunch it later with the same mount point, you wouldn't need to relaunch the contract-- the database has persisted the tokens contract.

Reply:
check